src/client/components/split-menu/split-menu.tsx

```diff
@@ -86,7 +86,7 @@ export class SplitMenu extends React.Component<SplitMenuProps, SplitMenuState> {
 
   globalKeyDownListener = (e: KeyboardEvent) => enterKey(e) && this.onOkClick();
 
-  saveGranularity = (granularity: string) => this.setState({ granularity });
+  saveGranularity = (granularity: string) => this.setState({ colors: Colors.fromLimit(this.props.dimension.name, 5), granularity });
```

Comment:
Looks suspicious. Why save new colours from limit when changing granularity?

Reply:
Well, if some colors are chosen (say `100-200, 300-400, 1000-1100`), what should be the chosen values after changing granularity to, let's say, `10`?